src/pages/Dictionary/Dictionary.jsx

```diff
@@ -1,32 +1,50 @@
 import { useSearchParams } from 'react-router-dom';
-import { fetchEntries, fetchTerms, fetchUsers } from '../../utils/fetchData';
+import { unpackEntriesQuery, unpackTermsQuery } from '../../utils/unpackQuery';
+import { getTermsEntriesStatus } from '../../utils/getTermsEntriesStatus';
+import getEntriesQuery from '../../utils/getEntriesQuery';
+import getTermsQuery from '../../utils/getTermsQuery';
+import CardLoading from './components/CardLoading';
 
 const Dictionary = () => {
   const [searchParams] = useSearchParams();
   const searchTerm = searchParams.get('term')?.toLowerCase();
 
-  const entries = fetchEntries();
-  const terms = fetchTerms();
-  const users = fetchUsers();
-  console.log('entries: ', entries, '\nterms: ', terms, '\nusers: ', users);
+  const termsQuery = getTermsQuery(searchTerm);
+  const { status: termsStatus, data: terms } = unpackTermsQuery(termsQuery);
+  const termid = searchTerm && terms ? Object.keys(terms)[0] : undefined;
```

Comment:
Is this relying on search terms?

Reply:
I think this component should only be showing a list of Entries.

Reply:
yes because we need to find the termid based on the search term.
If searchTerm is undefined, it means that a user is on the very first home page without searching anything, so we want to fetch all data. Im not sure if i got your question right tho 😅

Reply:
Ah! That does answer my question! If that is the case, why are we setting termid to undefined if we don't have a searchTerm?

Reply:
No! if termid is undefined, it will return a full list of entries.`getEntriesQuery({ termid: undefined });` is the same as `getEntriesQuery();`
This is a reference (https://tanstack.com/query/v4/docs/framework/react/guides/query-keys). useFirebaseQuery is using useQuert under the hood.
<img width="805" alt="image" src="https://github.com/rainawan/BruinDictionary/assets/96547645/7851d3c6-c0a6-41a6-b4c9-a73aeb127330">

Reply:
Also, in getEntriesQuery, if termid is undefined, it will not add the where function
<img width="411" alt="image" src="https://github.com/rainawan/BruinDictionary/assets/96547645/e735fbc6-15c5-4e74-a431-3236f8f9566c">

Reply:
ah i found i need to change queryKey